src/repository/ProductDatabaseRepository.js

```diff
@@ -6,6 +6,13 @@ import type { IBaseDatabase, IProductRepository, Product } from '../types';
 import COLLECTION_NAMES from './collectionNames';
 import BaseRepository from './BaseRepository';
 
+const getProductIDFromIDOrSlug = (IDOrSlug: string): ?number => {
+  const numericStringValue = IDOrSlug.replace(/[^0-9]/g, '');
+  return numericStringValue.length === IDOrSlug.length
```

Comment:
This won't work when you name your product something with a number in it.

I'd prefer that you just do a `try/catch` around a `parseInt` or check to see if isNaN or something.